HeterogeneousCore/AlpakaTest/test/testAlpakaModules.sh

```diff
@@ -5,21 +5,26 @@ function die { echo Failed $1: status $2 ; exit $2 ; }
 TEST_DIR=${LOCALTOP}/src/HeterogeneousCore/AlpakaTest/test
 
 if [ "$#" != "1" ]; then
-    die "Need exactly 1 argument ('cpu', 'cuda'), got $#" 1
+    die "Need exactly 1 argument ('cpu', 'cuda', or 'rocm'), got $#" 1
 fi
-if [ "$1" = "cuda" ]; then
-    TARGET=cuda
-elif [ "$1" = "cpu" ]; then
-    # In non-_GPU_ IBs, if CUDA is enabled, run the GPU-targeted tests
+if [[ "$1" =~ ^(cpu|cuda|rocm)$ ]]; then
+    TARGET=$1
+else
+    die "Argument needs to be 'cpu', 'cuda', or 'rocm'; got '$1'" 1
+fi
+
+# Some of the CPU-only tests fail if run on machine with GPU
+if [ "$TARGET" == "cpu" ]; then
     cudaIsEnabled
-    CUDA_ENABLED=$?
-    if [ "${CUDA_ENABLED}" = "0" ]; then
-        TARGET=cuda
-    else
-        TARGET=cpu
+    if [ "$?" == "0" ]; then
+        echo "Test target is 'cpu', but NVIDIA GPU is detected. Ignoring the CPU tests."
+        exit 0
+    fi
+    rocmIsEnabled
+    if [ "$?" == "0" ]; then
+        echo "Test target is 'cpu', but AMD GPU is detected. Ignoring the CPU tests."
+        exit 0
```

Comment:
In the test runner script I still ended up with this logic to not do anything if the `testHeterogeneousCoreAlpakaTestModulesCPU` test is run on a machine with a GPU (because tests on the automatic backend selection would fail). Is this the best way to go, or would there be some `scram` magic to achieve the same? @smuzaffar 

I would expect this use case to not to be common (so I'd be fine if we keep the logic here).

Reply:
@makortel , there is no need to run/check `{cuda|rocm}IsEnabled` in side the script. The https://github.com/cms-sw/cmssw-config/pull/95 change already takes care of it. `scram` only runs the tests for `nvidia` GPU if tests directly depend on `cuda` and `cudaIsEnabled` runs successfully. Same for `rocm` tests. As in the BuildFile you already have the correct deps so the script can be cleanup

Reply:
> `scram` only runs the tests for `nvidia` GPU if tests directly depend on `cuda` and `cudaIsEnabled` runs successfully. Same for `rocm` tests. As in the BuildFile you already have the correct deps so the script can be cleanup

Right, and that works great for the `testAlpakaModules.sh cuda` and `testAlpakaModules.sh rocm` tests. The `testAlpakaModules.sh cpu` case is different in the way that it succeeds only if the node has no GPU. In the GPU_X IB and GPU PR tests, where the `USER_UNIT_TESTS=cuda` is set explicitly the `testAlpakaModules.sh cpu` case gets properly ignored. But it will get run if a user runs just `scram b runtests` on a machine with a GPU. And that is the case where I'd want this test to be effectively skipped.

(as I mentioned earlier, I'd expect this use case to be rare, and am not sure if it is worth of the effort to be able to express this special case in the `BuildFile.xml`)

Reply:
> `scram` only runs the tests for `nvidia` GPU if tests directly depend on `cuda` and `cudaIsEnabled` runs successfully. Same for `rocm` tests. As in the BuildFile you already have the correct deps so the script can be cleanup

Right, and that works great for the `testAlpakaModules.sh cuda` and `testAlpakaModules.sh rocm` tests. The `testAlpakaModules.sh cpu` case is different in the way that it succeeds only if the node has no GPU. In the GPU_X IB and GPU PR tests, where the `USER_UNIT_TESTS=cuda` is set explicitly the `testAlpakaModules.sh cpu` case gets properly ignored. But it will get run if a user runs just `scram b runtests` on a machine with a GPU. And that is the case where I'd want this test to be effectively skipped.

(as I mentioned earlier, I'd expect this use case to be rare, and am not sure if it is worth of the effort to be able to express this special case in the `BuildFile.xml`)